Plugin~/WebRTCPlugin/GpuMemoryBufferPool.cpp

```diff
@@ -35,44 +35,40 @@ namespace webrtc
         auto it = resourcesPool_.begin();
         while (it != resourcesPool_.end())
         {
-            FrameReources* resources = it->get();
+            FrameResources* resources = it->get();
             if (!resources->IsUsed() && AreFrameResourcesCompatible(resources, size, format))
             {
                 resources->MarkUsed();
-                // copy texture
                 static_cast<GpuMemoryBufferFromUnity*>(resources->buffer_.get())->CopyBuffer(ptr);
                 return resources->buffer_;
             }
-            else
-            {
-                it++;
-            }
+            it++;
         }
-
-        rtc::scoped_refptr<GpuMemoryBufferInterface> buffer =
+        rtc::scoped_refptr<GpuMemoryBufferFromUnity> buffer =
             new rtc::RefCountedObject<GpuMemoryBufferFromUnity>(device_, ptr, size, format);
-        std::unique_ptr<FrameReources> resources = std::make_unique<FrameReources>(buffer);
+        std::unique_ptr<FrameResources> resources = std::make_unique<FrameResources>(buffer);
+        resources->MarkUsed();
+        buffer->CopyBuffer(ptr);
```

Comment:
`CopyBuffer` called in  `GpuMemoryBufferFromUnity` constructor.
Do you need to remove it?